symbolic-ppdb/tests/test_ppdb.rs

```diff
@@ -73,3 +86,77 @@ fn test_embedded_sources_with_metadata_broken() {
     let items = iter.collect::<Result<Vec<_>, _>>().unwrap();
     assert_eq!(items.len(), 0);
 }
+
+#[test]
+fn test_matching_ids() {
+    let pdb_buf = std::fs::read(fixture("windows/portable.pdb")).unwrap();
+    let pdb = PortablePdb::parse(&pdb_buf).unwrap();
+    let pdb_debug_id = pdb.pdb_id().unwrap();
+
+    let pe_buf = std::fs::read(fixture("windows/integration.dll")).unwrap();
+    let pe = PeObject::parse(&pe_buf).unwrap();
+    let pe_debug_id = pe.debug_id();
+
+    assert_eq!(pe_debug_id, pdb_debug_id);
+}
+
+#[test]
+fn test_pe_embedded_ppdb_without_sources() {
+    let pe_buf = std::fs::read(fixture(
+        "windows/Sentry.Samples.Console.Basic-embedded-ppdb.dll",
+    ))
+    .unwrap();
+    let pe = PeObject::parse(&pe_buf).unwrap();
+
+    let embedded_ppdb = pe.embedded_ppdb().unwrap().unwrap();
+
+    let mut ppdb_buf = vec![0; embedded_ppdb.get_size()];
+    embedded_ppdb.decompress(&mut ppdb_buf).unwrap();
```

Comment:
It does not matter for the unit test, but `Vec::with_capacity` allocates, but does not zero-initialize.

Other than that, writing into an `&mut` out paramater might be useful in theory, but I’m wondering if that is a case of YAGNI here, as I can’t imagine any other use-case than doing this with a fresh `Vec`. If the param would be a `impl Write`, that would be different, as you could write directly to a file. But this way, it might just be simpler to return `Vec<u8>` from that fn.

Reply:
Oh, thats why its zero-initialized I guess? Well even more reason to return a `Vec<u8>` and remove that footgun.

Reply:
ok changed